src/providers/on-chain-quote-provider.ts

```diff
@@ -971,32 +987,30 @@ export class OnChainQuoteProvider implements IOnChainQuoteProvider {
     }
   }
 
+  /**
+   * Throw an error for incorrect routes / funcitonName combinations
+   * @param routes Any combination of V3, V2, and Mixed routes.
+   * @param functionName
+   * @param usesMixedRouteQuoter Boolean indicating whether the mixedRouteQuoter needs to be used (for pure V2 and Mixed routes)
+   */
   protected validateRoutes(
-    routes: TRoute[],
+    routes: (V3Route | V2Route | MixedRoute)[],
     functionName: string,
-    isMixedRoutes: boolean
+    usesMixedRouteQuoter: boolean
   ) {
-    // we cannot have both V3 and MixedRoutes in the same call
+    // we cannot have both V3 and MixedRoutes or V2Routes in the same call since they use different quoter contracts
     if (
       routes.some((route) => route instanceof V3Route) &&
-      routes.some((route) => route instanceof MixedRoute)
+      usesMixedRouteQuoter
     ) {
       throw new Error(
-        'Cannot have both V3 and MixedRoutes in the same call to on chain Quoter'
+        'Cannot have mix and match V3 on chain quotes with MixedRoutes and/or V2Routes'
       );
     }
```

Comment:
Not sure I follow this. In my mind there are just 2 conditions:

1/ 'exactOut' must go on original quoter (which is covered below) 
2/ mixedRoutes must go on new quoter

Everything else can go on either?

Reply:
V2Routes cannot be run on old Quoter though

Reply:
Because we decided to just pass in one instance of `OnChainQuoteProvider` we now have to check at run time which quoter to pass the routes into. And I think the following should happen:
- We use the old quoter `QuoterV2` for all V3Routes (even though mixed route quoter will support V3Routes it will only support inputOnly routes)
- We use the new quoter `QuoterV3` (the mixed route quoter) for MixedRoutes and V2Routes that are exactInput (imo this second condition is still a little strange because the `QuoterV3` was written with the intention of just doing mixed routes, so we didn't consider exposing on chain quotes for v2 and thus, can now only do v2 exactInputs but I think its ok)